src/App.js

```diff
@@ -1,5 +1,146 @@
+const { Console, Random } = require('@woowacourse/mission-utils');
+const { MESSAGES, NUMBERS, UNITS, REGEX, CHOICE } = require('./constants');
+
 class App {
-  play() {}
+	computerNumArr = [];
+
+	play() {
+		Console.print(MESSAGES.START);
+		this.initGame();
+		this.playGame();
+	}
+
+	initGame() {
+		this.computerNumArr = [];
+
+		while (this.computerNumArr.length < 3) {
+			let number = Random.pickNumberInRange(1, 9);
+			if (!this.computerNumArr.includes(number))
+				this.computerNumArr.push(number);
+		}
+
+		return this.computerNumArr;
+	}
+
+	playGame() {
+		Console.readLine(
+			MESSAGES.REQUIRE_USER_INPUT(NUMBERS.GAME_MAX),
+			userAnswerStr => {
+				this.isValidNumber(userAnswerStr);
+
+				const [ball, strike] = this.compareComputerWithUser(
+					this.computerNumArr,
+					userAnswerStr
+				);
+				const resultMessage = this.convertScoreToMessage([
+					ball,
+					strike,
+				]);
+
+				this.showResultMessage(resultMessage);
+			}
+		);
+	}
+
+	isValidNumber(userAnswerStr) {
+		const answer = userAnswerStr.replace(REGEX.SPACE, '');
+		const { length } = answer;
+
+		if (length >= 4) throw new Error(MESSAGES.INVALID_LENGTH);
+
+		if (isNaN(answer)) throw new Error(MESSAGES.NOT_A_NUMBER);
+
+		if (this.removeDuplicated(userAnswerStr) !== userAnswerStr)
+			throw new Error(MESSAGES.DUPLICATED_NUM);
+
+		return true;
+	}
+
+	removeDuplicated(s) {
+		let answer = '';
+
+		for (let i = 0; i < s.length; i++) {
+			if (s.indexOf(s[i]) === i) answer += s[i];
+		}
+
+		return answer;
+	}
+
+	isValidChoice(playerChoice) {
+		if (!REGEX.CHOICE.test(playerChoice)) {
+			Console.print(MESSAGES.FORMAT_ERROR_CHOICE);
+			throw Error(MESSAGES.FORMAT_ERROR_CHOICE);
+		}
+		return true;
+	}
+
+	compareComputerWithUser(computerNumArr, userAnswerStr) {
+		let [ball, strike] = [0, 0];
+
+		userAnswerStr.split('').forEach((userAnswerStr, userAnswerIdx) => {
+			this.isValidNumber(userAnswerStr);
+
+			computerNumArr.map((computerNum, computerNumIdx) => {
+				if (
+					computerNum === Number(userAnswerStr) &&
+					computerNumIdx === userAnswerIdx
+				) {
+					strike++;
+					return;
+				}
+				if (computerNum === Number(userAnswerStr)) ball++;
+			});
+		});
+
+		return [ball, strike];
+	}
+
+	convertScoreToMessage([ball, strike]) {
+		let resultMessage = '';
+
+		if (strike > 0 && ball > 0) {
+			resultMessage = `${ball}${UNITS.BALL} ${strike}${UNITS.STRIKE}`;
+		} else if (strike > 0) {
+			resultMessage = `${strike}${UNITS.STRIKE}`;
+		} else if (ball > 0) {
+			resultMessage = `${ball}${UNITS.BALL}`;
+		} else {
+			resultMessage = UNITS.NOTHING;
+		}
```

Comment:
```suggestion
		if (strike > 0 && ball > 0) {
			return `${ball}${UNITS.BALL} ${strike}${UNITS.STRIKE}`;
		}
		if (strike > 0) {
			return `${strike}${UNITS.STRIKE}`;
		}
		if (ball > 0) {
			return `${ball}${UNITS.BALL}`;
		}
		return UNITS.NOTHING;
```

클린코드에서는 [else 키워드를 지양](https://www.google.com/search?q=%ED%81%B4%EB%A6%B0%EC%BD%94%EB%93%9C+else+%EC%A7%80%EC%96%91&oq=%ED%81%B4%EB%A6%B0%EC%BD%94%EB%93%9C+else+%EC%A7%80%EC%96%91)합니다. 이렇게 바꿔보는건 어떨까요?

Reply:
else 키워드가 지양되는 것은 알고 있으나 무조건 사용하지 말라는 것이 아닌 필요한 경우에는 사용한다고 알고 있어서요! 이부분은 로직상 else 를 제거하면 다르게 동작되기 때문에 남겨두었습니다 :)